src/main/java/dataAnalyzer/MovingAverage.java

```diff
@@ -0,0 +1,108 @@
+package dataAnalyzer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Calendar;
+
+public class MovingAverage implements TrendAnalyzer{
+    @Override
+    public List<PriceData> calculateMovingAverage(List<Price> priceList) {
+        // Implementiere die Berechnung des gleitenden Durchschnitts (Moving Average)
+        // Gib den Durchschnitt zu jedem Zeitpunkt als Liste von Double-Wert zurück
+
+	List<PriceData> priceAverages = new ArrayList<>();
+    List<Double> previousPrices = new ArrayList<>();
+
+    for (Price price : priceList) {
+        previousPrices.add((price.getlow() + price.gethigh() + price.getopen() + price.getclose()) / 4);
+
+        if (previousPrices.size() > 200) {
+            previousPrices.remove(0); // Entferne den ältesten Preis, wenn die Liste länger als 200 ist
+        }
+
+        double average = calculateAverage(previousPrices);
+        Calendar date = price.getday();
+
+        PriceData priceData = new PriceData(average, date);
+        priceAverages.add(priceData);
+    }
+
+    return priceAverages;
+}
+
+	private double calculateAverage(List<Double> prices) {
+    	double sum = 0.0;
+    	for (double price : prices) {
+        	sum += price;
+    	}
+    	return sum / prices.size();
+	}
+
+
+	@Override
+	public List<Boolean> AverageIntersectsStock(List<PriceData> priceAverages, List<Price> priceList){
+		List<Boolean> intersections = new ArrayList<>();
+	
+		 //Implementiert eine Funktion welche Testet, ob der Moving Average für einen Zeitpunkt gleich dem Stockpreis ist
+		 //Gibt die Ergebnise als Liste von Booleans zurück
+
+		for (int i = 0; i <= priceAverages.size(); i++) {
+			double average = priceAverages.get(i).getAverage();
+			double graphLow = priceList.get(i).getlow();
+			double graphHigh = priceList.get(i).gethigh();
+			double graphOpen = priceList.get(i).getopen();
+			double graphClose = priceList.get(i).getclose();
+	
+			
+			if(average==graphLow|| average == graphHigh|| average==graphClose|| average==graphOpen){
```

Comment:
Do we want to check whether the recorded prices meet (like we do here) or do we want to check whether the average intersected the price at some point during the interval, which is represented in the `Price` structure? The latter would mean, that we could simply do the following check:
```java
average >= graphLow && average <= graphHigh
```